beacon-chain/rpc/validator_server.go

```diff
@@ -294,10 +279,111 @@ func (vs *ValidatorServer) ValidatorStatus(
 	return res, nil
 }
 
-func (vs *ValidatorServer) validatorStatus(pubkey []byte, beaconState *pbp2p.BeaconState) (pb.ValidatorStatus, error) {
+func (vs *ValidatorServer) MultipleValidatorStatus(
+	ctx context.Context,
+	pubkeys [][]byte) (bool, []*pb.ValidatorActivationResponse_Status, error) {
+
+	activeValidatorExists := false
+
+	beaconState, err := vs.beaconDB.HeadState(ctx)
+	if err != nil {
+		return false, nil, fmt.Errorf("could not fetch beacon state: %v", err)
+	}
+
+	validatorMap := stateutils.ValidatorIndexMap(beaconState)
+	statusResponses := make([]*pb.ValidatorActivationResponse_Status, len(pubkeys))
+
+	for i, key := range pubkeys {
+		statusResponses[i] = &pb.ValidatorActivationResponse_Status{
+			PublicKey: key,
+			Status:    &pb.ValidatorStatusResponse{},
+		}
+		dep, eth1BlockNumBigInt, err := vs.beaconDB.DepositByPubkey(ctx, key)
+		if err != nil {
+			return activeValidatorExists, nil, err
+		}
+		if eth1BlockNumBigInt == nil {
+			statusResponses[i].Status = &pb.ValidatorStatusResponse{
+				Status:                 pb.ValidatorStatus_UNKNOWN_STATUS,
+				ActivationEpoch:        params.BeaconConfig().FarFutureEpoch - params.BeaconConfig().GenesisEpoch,
```

Comment:
why subtract `GenesisEpoch` here?

Reply:
oh this was for readability, following on from this PR #2475

Reply:
aha, got it. thanks!